mllib/src/main/scala/org/apache/spark/ml/Estimator.scala

```diff
@@ -79,7 +82,52 @@ abstract class Estimator[M <: Model[M]] extends PipelineStage {
    */
   @Since("2.0.0")
   def fit(dataset: Dataset[_], paramMaps: Array[ParamMap]): Seq[M] = {
```

Comment:
Let's deprecate this method, say it will be removed in the 3.0.0 release, and tell users and developers to use fitMultiple instead.

Reply:
Seemingly we should use the following note to deprecate it:
"""
.. note:: Deprecated in 2.3.0. Use :func:`Estimator.fitMultiple` instead.
"""
and like other places, add a warning
```
warnings.warn("Deprecated in 2.3.0. Use Estimator.fitMultiple instead.", DeprecationWarning)
```
?